Core/Src/processing.c

```diff
@@ -0,0 +1,365 @@
+/**
+ * @file processing.c
+ * @author Scott Abramson
+ * @brief Definitions for tasks for processing data.
+ * @version 0.1
+ * @date 2024-08-04
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#include "processing.h"
+#include "state_machine.h"
+#include "queues.h"
+#include "cerb_utils.h"
+#include "nero.h"
+#include "can_handler.h"
+#include "cerberus_conf.h"
+#include "dti.h"
+#include "queues.h"
+#include "serial_monitor.h"
+#include "bms.h"
+#include "emrax.h"
+#include "monitor.h"
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+#include <math.h>
+#include <stdlib.h>
+
+/* DO NOT ATTEMPT TO SEND TORQUE COMMANDS LOWER THAN THIS VALUE */
+#define MIN_COMMAND_FREQ  60 /* Hz */
+#define MAX_COMMAND_DELAY 1000 / MIN_COMMAND_FREQ /* ms */
+
+static float torque_limit_percentage = 1.0;
+
+#define TSMS_DEBOUNCE_PERIOD 500 /* ms */
+
+static bool tsms = false;
+osMutexId_t tsms_mutex;
+
+bool get_tsms()
+{
+	bool temp;
+	osMutexAcquire(tsms_mutex, osWaitForever);
+	temp = tsms;
+	osMutexRelease(tsms_mutex);
+	return temp;
+}
+
+void tsms_debounce_cb(void *arg)
+{
+	/* Set TSMS state to new debounced value */
+	osMutexAcquire(tsms_mutex, osWaitForever);
+	tsms = *((bool *)arg);
+	osMutexRelease(tsms_mutex);
+	/* Tell NERO allaboutit */
+	send_nero_msg();
```

Comment:
again why are we sending a can message based on a hw time we gonna flood everything with a bunch of nero messages.  Can we either just do it in reaction to one thing (ex, default task or set_mph) or just throw it into a thread?

Reply:
Nero state updates won't flood can cuz they happen infrequently

Reply:
Upon every dti message it's sent (25ms), every tsms update (100ms?) and in default task (500ms).  That's more than any other message on the car except button IO.